Sources/Sentry/SentryFileManager.m

```diff
@@ -696,19 +728,89 @@ - (void)createPathsWithOptions:(SentryOptions *)options
     self.envelopesPath = [self.sentryPath stringByAppendingPathComponent:EnvelopesPathComponent];
 }
 
-- (BOOL)createDirectoryIfNotExists:(NSString *)path error:(NSError **)error
-{
-    if (![[NSFileManager defaultManager] createDirectoryAtPath:path
-                                   withIntermediateDirectories:YES
-                                                    attributes:nil
-                                                         error:error]) {
-        *error = NSErrorFromSentryErrorWithUnderlyingError(kSentryErrorFileIO,
-            [NSString stringWithFormat:@"Failed to create the directory at path %@.", path],
-            *error);
-        return NO;
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+/**
+ * @note This method must be statically accessible because it will be called during app launch,
+ * before any instance of @c SentryFileManager exists, and so wouldn't be able to access this path
+ * from an objc property on it like the other paths.
+ */
+NSString *
+sentryApplicationSupportPath(void)
+{
+    static NSString *sentryApplicationSupportPath;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSArray *paths = NSSearchPathForDirectoriesInDomains(
+            NSApplicationSupportDirectory, NSUserDomainMask, YES);
+        NSString *applicationSupportDirectory = [paths firstObject];
```

Comment:
`h`: I'm unsure if it could happen that the above statement doesn't return a list of paths. Maybe in some restricted environments, that could happen. I think it's better to assume it could. If it does then we pass `nil` to `access` in `appLaunchProfileConfigFileExists`, and I'm unsure what the behavior would be. As this code will run for everybody, we should be extra careful

Reply:
good call on checking for `nil` before passing a parameter to `access` 👍🏻